### PR TITLE
RAC-5682 - Redfish Compliance, correct systems/ethernetinterface odata links for non-dell

### PR DESCRIPTION
**Pull request by larry-dean**
* Create valid odtata linkes for non-dell devices
* Added veiw for ethernetintercae/index for non-dell devices
* Correct spec tests to allow ethernetinterface for non-dell devices

### CODE REVIEW

**Comment by larry-dean** [comment]
jenkins: depends on https://github.com/RackHD/RackHD/pull/841

**Comment by larry-dean** [comment]
test this please

File: spec/lib/api/redfish-1.0/system-spec.js [hunk 1]
```diff
@@ -789,30 +821,53 @@ describe('Redfish Systems Root', function () {
...
+                expect(tv4.validate.called).to.be.true;
+                expect(validator.validate.called).to.be.true;
+                expect(redfish.render.called).to.be.true;
             });
     });
 
-    it('should 404 a valid identifier for ethernet index query with invalid index', function() {
-        return helper.request().get('/redfish/v1/Systems/' + dellNode.id + '/EthernetInterfaces/' + "BADNIC.Integrated.1-1-1")
+
+    it('should 404 a non-delll with valid index and invalid ethernet index', function() {
```
Comment (jimturnquist) [inline spec/lib/api/redfish-1.0/system-spec.js]: delll should be dell

File: spec/lib/api/redfish-1.0/system-spec.js [hunk 2]
```diff
@@ -767,13 +805,7 @@ describe('Redfish Systems Root', function () {
             .expect(404);
     });
 
-    it('should 404 a non-Dell identifier for ethernet query', function() {
```
Comment (jimturnquist) [inline spec/lib/api/redfish-1.0/system-spec.js]: Was this test deleted?

File: spec/lib/api/redfish-1.0/system-spec.js [hunk 3]
```diff
@@ -789,30 +821,53 @@ describe('Redfish Systems Root', function () {
             });
     });
 
-    it('should 404 an invalid identifier for ethernet index query with valid index', function() {
```
Comment (jimturnquist) [inline spec/lib/api/redfish-1.0/system-spec.js]: It looks like some of the tests were replaced.  Was this the intent?

File: spec/lib/api/redfish-1.0/system-spec.js (hunk 2 above)
Comment (larry-dean) [inline spec/lib/api/redfish-1.0/system-spec.js]: yes, a duplicate

File: spec/lib/api/redfish-1.0/system-spec.js (hunk 1 above)
Comment (larry-dean) [inline spec/lib/api/redfish-1.0/system-spec.js]: will fix

**Comment by hohene** [comment]
This PR depends on the RackHD PR: https://github.com/RackHD/RackHD/pull/841 
and is causing failures.  not sure how depends is supposed to work in this case.

**Pull request merged**